src/dex/balancer-v2/balancer-v2.ts

```diff
@@ -659,19 +745,34 @@ export class BalancerV2
       count,
     };
 
-    const query = `query ($tokens: [Bytes!], $count: Int) {
-      pools (first: $count, orderBy: totalLiquidity, orderDirection: desc,
-           where: {tokensList_contains: $tokens,
-                   swapEnabled: true,
+    const query = `
+      query ($tokens: [Bytes!], $count: Int) {
+          pools: pools (first: $count, orderBy: totalLiquidity, orderDirection: desc, 
+           where: {tokensList_contains: $tokens, 
+                   swapEnabled: true, 
                    totalLiquidity_gt: 0}) {
-        address
-        totalLiquidity
-        tokens {
-          address
-          decimals
-        }
-      }
-    }`;
+            address
+            totalLiquidity
+            tokens {
+              address
+              decimals
+            }
+          }
+
+          poolsAll: pools(first: 1000, orderBy: totalLiquidity, orderDirection: desc, where: {swapEnabled: true, poolType_in: ["MetaStable", "Stable", "Weighted", "LiquidityBootstrapping", "Investment", "StablePhantom", "AaveLinear", "ERC4626Linear"]}) {
```

Comment:
VirtualBoostedPools creation needs info of all pools. VirtualBoostedPools could be taken from `this.eventPools.virtualBoostedPools` but in `balancer-v2-integration.test.ts` the `initializePricing` function isn't called so pools haven't been pre-loaded. Wasn't sure if this was a requirement.

Reply:
Yes that is a requirement. But you can implement `updatePoolState` this is function that will be called  before `getTopPoolsForToken`. `updatePoolState` is called once before calling `getTopPoolsForToken` multiple times for different tokens. So you can fetch and save the state in the memory and reuse it in `getTopPoolsForToken`. I know this part is a bit confusing as you only know the opensource codebase but it is just a side effect of merging multiple dex interfaces into one

Reply:
Implemented `updatePoolState` which fetches all pools from Subgraph, adds VirtualBoostedPools and stores in memory to be reused by `getTopPoolsForToken`